Fix/18885 migrate avatar token

thebinij opened the pull request:
## Explanation
* Closes #18885 
<!--
Thanks for the pull request. Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:

* What is the current state of things and why does it need to change?
* What is the solution your changes offer and how does it work?

Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:

* Fixes #12345
* See: #67890
-->

## Screenshots/Screencaps

<!-- If you're making a change to the UI, make sure to capture a screenshot or a short video showing off your work! -->

### Before

<!-- How did the UI you changed look before your changes? Drag your file(s) below this line: -->
https://github.com/MetaMask/metamask-extension/assets/93417120/351d02e4-6b55-418a-9505-5fc5699ac026

### After

<!-- How does it look now? Drag your file(s) below this line: -->
https://github.com/MetaMask/metamask-extension/assets/93417120/4a884625-9d50-4eda-85c7-79c2b82578a8

## Manual Testing Steps
No functional changes
<!--
How should reviewers and QA manually test your changes? For instance:

- Go to this screen
- Do this
- Then do this
-->

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

thebinij committed:
Migrate AvatarToken

fix lint

fixing error

replace BoxTextAlign with TextAlign

using AvatarTokenSize

fix avatar-base-type

georgewrmarshall commented on ui/components/component-library/avatar-token/avatar-token.tsx:
```diff
@@ -0,0 +1,85 @@
+    return (
+      <AvatarBase
+        ref={ref}
+        size={size as unknown as AvatarBaseSize}
```
Change back to size and add `AvatarTokenSize` as a union type to `AvatarBase`
```suggestion
        size={size}
```

thebinij replied:
Agreed, let's do that.

georgewrmarshall reviewed (commented):
Hey @thebinij, I've checked out your branch and tried a few things to get the types to work but I can't seem to find a reliable pattern. I think for now we just make `AvatarBaseProps` accept all variant size types as a union type for now and update it in the future once we find a scalable pattern to use.

georgewrmarshall commented on ui/components/component-library/avatar-token/avatar-token.test.tsx:
```diff
@@ -19,42 +19,66 @@ describe('AvatarToken', () => {
   it('should render correctly', () => {
     const { getByTestId, container } = render(
-      <AvatarToken {...args} data-testid="avatar-token" />,
+      <AvatarToken children={undefined} {...args} data-testid="avatar-token" />,
```
Can we remove children for the types?

georgewrmarshall commented:
Hey @thebinij, Thanks for your time and effort in creating this pull request. As you're probably aware we have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at https://github.com/MetaMask/metamask-extension/issues/19239 and a draft pull request at https://github.com/MetaMask/metamask-extension/pull/19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

garrettbear committed:
Migrate AvatarToken

fix lint

fixing error

replace BoxTextAlign with TextAlign

using AvatarTokenSize

fix avatar-base-type

Added union AvatarTokenSize

TokenProps extends BaseProps

remove children prop from BaseProps

replace deprecated

lint fix

garrettbear committed:
update TS types

garrettbear committed:
Merge branch 'develop' into fix/18885-Migrate-AvatarToken

garrettbear committed:
fix types and children issue

garrettbear committed:
Merge branch 'develop' into fix/18885-Migrate-AvatarToken

georgewrmarshall commented on ui/components/component-library/avatar-network/avatar-network.types.ts:
```diff
@@ -13,7 +13,7 @@ export enum AvatarNetworkSize {
  */
 export interface AvatarNetworkStyleUtilityProps
-  extends Omit<AvatarBaseStyleUtilityProps, 'size'> {
+  extends Omit<AvatarBaseStyleUtilityProps, 'size' | 'children'> {
```
I think this makes sense with my limited TypeScript knowledged brain. We are adjusting the types at the same level we are increasing the specificity of the component.

georgewrmarshall reviewed (approved):
Looks alll G to ME! Thanks for your contribution @thebinij and for supporting @garrettbear 💯

georgewrmarshall commented:
> `ButtonLink` lint error that is probably coming from the ButtonBase update and hopefully will be fixed when converting ButtonLink to TS.
> 
> <img alt="Screenshot 2023-07-25 at 3 59 55 PM" width="758" src="https://user-images.githubusercontent.com/26469696/256057372-bb6730d5-d867-4fc6-a839-13fca7f70052.png">

Looks like we are safe for the most part as the only use in tsx files are in stories which are ignored from the linter. I think we can prioritize buttons next

![Screenshot 2023-07-25 at 5 29 05 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/fb11ff3a-f8cc-45b1-9053-15a62f3a7f88)
